--- a/absl/meta/type_traits.h
+++ b/absl/meta/type_traits.h
@@ -503,7 +503,13 @@
 //
 // According to https://github.com/abseil/abseil-cpp/issues/1479, this does not
 // work with NVCC either.
-#if ABSL_HAVE_BUILTIN(__is_trivially_relocatable) &&                 \
+#if ABSL_HAVE_BUILTIN(__builtin_is_cpp_trivially_relocatable) &&      \
+    !(defined(__clang__) && (defined(_WIN32) || defined(_WIN64))) &&  \
+    !defined(__NVCC__)
+template <class T>
+struct is_trivially_relocatable
+    : std::integral_constant<bool, __builtin_is_cpp_trivially_relocatable(T)> {};
+#elif ABSL_HAVE_BUILTIN(__is_trivially_relocatable) &&                 \
     !(defined(__clang__) && (defined(_WIN32) || defined(_WIN64))) && \
     !defined(__NVCC__)
 template <class T>
